probe-dictionary/explore.js

```diff
@@ -431,24 +431,34 @@ function shortVersion(v) {
   return v.split(".")[0];
 }
 
-function friendlyRecordingRange(firstVersion, expiry) {
-  if (expiry == "never") {
-    return `from ${firstVersion}`;
+function friendlyRecordingRangeForHistory(history, channel, filterPrerelease) {
+  const last = array => array[array.length - 1];
+
+  if ((channel == "release") && filterPrerelease) {
+    history = history.filter(h => h.optout);
   }
-  return `${firstVersion} to ${parseInt(shortVersion(expiry)) - 1}`;
-}
 
-function friendlyRecordingRangeForState(state, channel) {
-  const firstVersion = getVersionRange(channel, state["revisions"]).first;
-  const expiry = state.expiry_version;
-  return friendlyRecordingRange(firstVersion, expiry);
-}
+  if (history.length == 0) {
```

Comment:
Why are we returning `"never"` if there's no history? Shouldn't this just be the case for `expiry == "never"`?

Reply:
The prior filter statement can result in an empty history (consider prerelease-only probes on "release" channel).
That should result in a "never" recording range on that channel.